experimental/xformer/Transforms/TranslateToCustomop.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright 2021 XMOS LIMITED. This Software is subject to the terms of the
+// XMOS Public License: Version 1
+
+#include "IR/XCoreOps.h"
+
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "tensorflow/compiler/mlir/lite/ir/tfl_ops.h"
+
+namespace mlir {
+namespace xcore {
+
+namespace {
+/// This pass translates XCore ops to TFLite custom ops.
+struct TranslateToCustomOp
+    : public PassWrapper<TranslateToCustomOp, FunctionPass> {
+  void runOnFunction() override;
+};
+
+template <typename XCoreOp>
+struct RewriteToCustomOp : public OpRewritePattern<XCoreOp> {
+  using OpRewritePattern<XCoreOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(XCoreOp xc_op,
+                                PatternRewriter &rewriter) const override {
+    std::vector<uint8_t> options = xc_op.buildCustomOptions();
+    Operation *op = xc_op.getOperation();
+    ShapedType type = RankedTensorType::get(
```

Comment:
Is there a reason for explicit types here? If not, consider:
```suggestion
    auto options = xc_op.buildCustomOptions();
    auto *op = xc_op.getOperation();
    auto type = RankedTensorType::get(
```